ffmpeg-plugin/mcm_video_tx.c

```diff
@@ -29,85 +28,102 @@ typedef struct McmVideoMuxerContext {
 
-    switch (param.payload_type) {
-    case PAYLOAD_TYPE_RTSP_VIDEO:
-    case PAYLOAD_TYPE_ST20_VIDEO:
-    case PAYLOAD_TYPE_ST22_VIDEO:
-        /* video format */
-        param.payload_args.video_args.width   = param.width = s->width;
-        param.payload_args.video_args.height  = param.height = s->height;
-        param.payload_args.video_args.fps     = param.fps = av_q2d(s->frame_rate);
-
-        switch (s->pixel_format) {
-        case AV_PIX_FMT_NV12:
-            param.pix_fmt = PIX_FMT_NV12;
-            break;
-        case AV_PIX_FMT_YUV422P:
-            param.pix_fmt = PIX_FMT_YUV422P;
-            break;
-        case AV_PIX_FMT_YUV444P10LE:
-            param.pix_fmt = PIX_FMT_YUV444P_10BIT_LE;
-            break;
-        case AV_PIX_FMT_RGB24:
-            param.pix_fmt = PIX_FMT_RGB8;
-            break;
-        case AV_PIX_FMT_YUV422P10LE:
-        default:
-            param.pix_fmt = PIX_FMT_YUV422P_10BIT_LE;
-        }
-
-        param.payload_args.video_args.pix_fmt = param.pix_fmt;
-        break;
-
-    default:
-        av_log(avctx, AV_LOG_ERROR, "Unknown payload type\n");
+    err = mcm_get_client(&s->mc);
+    if (err) {
+        av_log(avctx, AV_LOG_ERROR, "Get mesh client failed: %s (%d)\n",
+               mesh_err2str(err), err);
         return AVERROR(EINVAL);
     }
 
-    s->tx_handle = mcm_create_connection(&param);
-    if (!s->tx_handle) {
-        av_log(avctx, AV_LOG_ERROR, "Create connection failed\n");
-        return AVERROR(EIO);
+    err = mesh_create_connection(s->mc, &s->conn);
+    if (err) {
+        av_log(avctx, AV_LOG_ERROR, "Create connection failed: %s (%d)\n",
+               mesh_err2str(err), err);
+        err = AVERROR(EIO);
+        goto exit_put_client;
+    }
+
+    err = mcm_parse_conn_param(avctx, s->conn, kind, s->ip_addr, s->port,
+                                  s->protocol_type, s->payload_type,
+                                  s->socket_name, s->interface_id);
+    if (err) {
+        av_log(avctx, AV_LOG_ERROR, "Configuration parsing failed: %s (%d)\n",
+               mesh_err2str(err), err);
+        err = AVERROR(EINVAL);
+        goto exit_delete_conn;
+    }
+
+    err = mesh_apply_connection_config_video(s->conn, &cfg);
+    if (err) {
+        av_log(avctx, AV_LOG_ERROR, "Failed to apply connection config: %s (%d)\n",
+               mesh_err2str(err), err);
+        err = AVERROR(EIO);
+        goto exit_delete_conn;
+    }
+
+    err = mesh_establish_connection(s->conn, kind);
+    if (err) {
+        av_log(avctx, AV_LOG_ERROR, "Cannot establish connection: %s (%d)\n",
+               mesh_err2str(err), err);
+        err = AVERROR(EIO);
+        goto exit_delete_conn;
     }
 
     av_log(avctx, AV_LOG_INFO,
            "w:%d h:%d pixfmt:%s fps:%d\n",
            s->width, s->height, av_get_pix_fmt_name(s->pixel_format),
            (int)av_q2d(s->frame_rate));
     return 0;
+
+exit_delete_conn:
+    mesh_delete_connection(&s->conn);
```

Comment:
Shouldn't we check the output of mesh_delete_connection here?

Reply:
This function call is already in the path of exiting due to an error happened before. So, the extra error checking here is redundant and could potentially be misleading as to the actual error occurred in the previous place in the code. I suggest leaving it here as it is.